### PR TITLE
fix: use a modern prompt to avoid ^[[D char when using arrow left and…

### PR DESCRIPTION
**Pull request by gabkk**
### Background
This PR aims to fix an issue with the input prompt where arrow keys would generate unwanted characters (`^[[D`) in the input when using arrows.

We also want to ensure that the colored output works correctly with the new input method.

prompt_toolkit also support history with up and down arrow which could be useful if we plan to have more user interaction with the agent in the futur

### Changes
1. Replaced the `input()` function with `prompt_toolkit`'s `PromptSession` to prevent unwanted characters when using arrow keys.
2. Adjusted all `clean_input` calls to use `prompt_toolkit`'s `HTML` formatting instead of `colorama` ANSI escape codes.
3. Added the `prompt_toolkit` package to the `requirements.txt` file.

### Documentation
The changes are documented in the form of in-code comments and by updating the `clean_input` function to accept `FormattedText` from `prompt_toolkit`.

### Test Plan
Tested the changes by running the script and interacting with the modified prompts. Verified that the arrow keys no longer produce unwanted characters and that the colored output is displayed correctly.

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Comment by hdkiller** [comment]
There is an another PR regarding this with readline. Why this is better than that? 

Check #2558

**Comment by gabkk** [comment]
> There is an another PR regarding this with readline. Why this is better than that?
> 
> Check #2558

Hi @hdkiller 

realine is already a good improvement but i believe that we will have much more interaction with auto-GPT using the terminal in the futur and readline is really basic.

prompt_toolkit allows:
- multi-line editing
- syntax highlighting
- input validation  (which can be really neat using gtp 3.5 turbo)
- auto-completion 
- creating complex user interfaces with colors, styles and widget like progress bars, menus, dialog box, html tags ect..
- mouse interaction

Which i believe will make the terminal UX much better.

I kept this PR small to know if you guys are willing to use prompt_toolkit. But i can create an other PR if this one is merged to show case how we can improve the current UI.

Some cool improvement could be to have options when auto-GPT want to execute a task, for example you want to do a web search Auto Gpt prompt display 3 option (BIng / google / duckduck go), you could use the arrow to choose between those 3 choices.

I think it will push the terminal interaction to an other level

File: autogpt/agent/agent.py [hunk 1]
```diff
@@ -127,9 +127,8 @@ def start_interaction_loop(self):
                     flush=True,
                 )
                 while True:
-                    console_input = clean_input(
-                        Fore.MAGENTA + "Input:" + Style.RESET_ALL
-                    )
+                    console_input = clean_input(HTML('<ansimagenta>Input:</ansimagenta>'))
```
Comment (ntindle) [inline autogpt/agent/agent.py]: I'd integrate this better similar to how it had constants before

File: autogpt/utils.py [hunk 2]
```diff
@@ -4,21 +4,23 @@
...
-# Use readline if available (for clean_input)
-try:
-    import readline
-except:
-    pass
+session = PromptSession(history=InMemoryHistory())
 
-
-def clean_input(prompt: str = ""):
+def clean_input(prompt: Union[str, FormattedText] = ""):
```
Comment (ntindle) [inline autogpt/utils.py]: Is this used elsewhere in the repo?

Comment (ntindle) [inline autogpt/utils.py]: Add tests for this if modifying it

**Review by ntindle — CHANGES_REQUESTED** [review]

File: autogpt/utils.py (hunk 2 above)
Comment (gabkk) [reply autogpt/utils.py]: yeah clean_input is used in 5 places they are all covered

Comment (gabkk) [reply autogpt/utils.py]: ok i ll add test now

**Comment by gabkk** [comment]
@ntindle 

Added tests, fixed the linter issue and add colors as constants to the utils file

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Comment by Boostrix** [comment]
> prompt_toolkit also support history with up and down arrow which could be useful if we plan to have more user interaction with the agent in the futur

FYI: early experiments using prompt-toolkit:
#4078 

agpt-ui2

**Comment by ariel-frischer** [comment]
I'm getting this `^[[D` issue when I press left arrow key. Hope this gets fixed soon.

**Comment by erik-megarad** [comment]
Simplest implementation to fix issue in #4649

**Comment by ariel-frischer** [comment]
> Simplest implementation to fix issue in #4649

You are the man, thank you will merge this.

**Comment by Pwuts** [comment]
Closing as fixed in #4649

**Pull request closed without merging**